openfe/protocols/openmm_rbfe/_rbfe_utils/topologyhelpers.py

```diff
@@ -130,39 +133,93 @@ def _remove_constraints(old_to_new_atom_map, old_system, old_topology,
     * Very slow, needs refactoring
     * Can we drop having topologies as inputs here?
     """
+    def get_the_H(A, B, C, D, oldH, newH) -> int:
```

Comment:
so a curveball towards generalisability here, what if we have a constrained bond that doesn't involve a hydrogen?

Reply:
For the current case, is there a way we could just check and raise an error? I don't particularly think we need to support this edge case just know to fail it.

Reply:
yeah ok, I can add this tomorrow